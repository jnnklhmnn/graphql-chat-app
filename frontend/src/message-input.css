.message-input {
  width: 100%;
  background-color:#a29a9a;
  border-radius: 0 0 4px 4px;
}

.input-wrapper {
  padding: 10px;
  display: flex;
  align-items: center;
}

.message-input-field {
  width: 60%;
  resize: none;
  padding: 5px;
  font-size: 16px;
}

.input-button {
  width: 28%;
  margin-left: 20px;
  height: 38px;
  font-size: 15px;
  box-shadow: none;
  background-color:#46b546;
  color: #ffffff;
  border: none;
}

.input-button:disabled {
  background-color: #94a59d;
}

