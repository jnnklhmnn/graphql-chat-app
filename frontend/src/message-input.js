import React, { Component } from "react";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import { ALL_POSTS_QUERY } from "./App.js";
import "./message-input.css";

export class MessageInput extends Component {
  state = {
    message: ""
  };

  componentDidMount() {
    const container = document.querySelector(".message-section");
    if (container) {
      container.scrollTop = container.scrollHeight;
    }
  }

  componentDidUpdate() {
    const container = document.querySelector(".message-section");
    if (container) {
      container.scrollTop = container.scrollHeight;
    }
  }

  /**
   * Sends request to mutate the Schema to add new Message to Database,
   * triggers refetch of ALL_POSTS_QUERY.
   */

  addMessage = async () => {
    const { message } = this.state;
    this.setState({ message: "" });
    await this.props.postMutation({
      variables: {
        message
      },
      refetchQueries: [{ query: ALL_POSTS_QUERY }]
    });
  };

  render() {
    return (
      <div className="message-input">
        <div className="input-wrapper">
          <textarea
            className="message-input-field"
            value={this.state.message}
            onChange={e => this.setState({ message: e.target.value })}
            type="text"
            placeholder="Enter your message here"
          />
          <button
            className="input-button"
            disabled={!this.state.message}
            onClick={() => this.addMessage()}
          >
            Send
          </button>
        </div>
      </div>
    );
  }
}

const POST_MUTATION = gql`
  mutation($message: String!) {
    createPost(message: $message) {
      id
      message
    }
  }
`;

export default graphql(POST_MUTATION, { name: "postMutation" })(MessageInput);
