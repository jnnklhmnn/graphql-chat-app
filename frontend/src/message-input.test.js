import { MessageInput } from './message-input.js';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

describe('MessageInput', () => {
    test('renders with data correctly', () => {
      Enzyme.configure({ adapter: new Adapter() });
      const wrapper = shallow(<MessageInput  />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  // should enable button on update state... der ist einfach.