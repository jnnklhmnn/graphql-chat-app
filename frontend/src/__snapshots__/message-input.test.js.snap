// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput renders with data correctly 1`] = `
<div
  className="message-input"
>
  <div
    className="input-wrapper"
  >
    <textarea
      className="message-input-field"
      onChange={[Function]}
      placeholder="Enter your message here"
      type="text"
      value=""
    />
    <button
      className="input-button"
      disabled={true}
      onClick={[Function]}
    >
      Send
    </button>
  </div>
</div>
`;
